hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -507,8 +507,8 @@ public static List<HoodieRecord> convertMetadataToFilesPartitionRecords(HoodieCl
       final String partition = getPartitionIdentifier(partitionName);
       // Files deleted from a partition
       List<String> deletedFiles = partitionMetadata.getDeletePathPatterns();
-      HoodieRecord record = HoodieMetadataPayload.createPartitionFilesRecord(partition, Option.empty(),
-          Option.of(new ArrayList<>(deletedFiles)));
```

Comment:
We were creating a copy of this list. I didn't see any reason to do this but let me know if I'm missing something.